57-git.Rmd

```diff
@@ -3,31 +3,220 @@
-o GitHub: Create an account, log in.
-o Demo: we show how to create a test repo (which has one or two files) and we add them to it to demonstrate
-o They can create an issue to show that they have access to the online repo, comment on it, link to one of the files, close an issue, etc.
+Understand the theoretical concepts of using Git and GitHub: repo (local and 
+remote), pull, push, commit, clone, conflicts, issues, pull request (which 
+prevents conflicts). Explain with a concept map 
+
+
+`r msmbstyle::question_begin()`
+
+1. Create an account on [GitHub](https://github.com/). Choose your username 
+wisely as you might want to reuse it later and will have to share it.
+**Should we ask them to share it on moodle or somewhere for the next exercise ?**
+
+2. Create your own repository and open an issue on it.
+
+`r msmbstyle::question_end()`
+
+`r msmbstyle::question_begin()`
+
+You have now been added to the WSBIM2122_GitHub_training repository.
+**We need to add 1 or 2 simple files to the repo**
+
+1. Reply to the first issue to verify that you're able to access the repo.
+
+2. Create a new issue where you make a link to a file or a specific line of a 
+file and assign it to yourself. 
+
+3. Close that issue to mark it as done.
+
+`r msmbstyle::question_end()`
+
 
 ## Git and RStudio
 
-Connect local RStudio with a repo: install git, create PAT and, create a 2-3 repo and clone (via Project)
+It is possible to use Git through the RStudio interface to manage everything 
+that is on a repository. This is useful if you're working mainly with R and 
+want to use Git version control but also if you want to do collaborative work. 
+The RStudio interface then allows a click-button way of using git that can be 
+easier than the traditional command line. 
+
+This section of the course is based on the *Happy Git and GitHub for the useR* 
+[tutorial](https://happygitwithr.com/) by Jennifer Bryan.
+
+### Install Git
+
+Git needs to be install on your computer so RStudio can call it, or to use it
+through the command line. Installation is different depending on the operating 
+system of your computer.
+
+- **Windows users**
+
+The easiest way is to install [Git for Windows](https://gitforwindows.org/), 
+also called "Git bash". This allows to get Git but also a Bash shell that is 
+useful to venture outside of R/RStudio. 
+
+When installing Git for Windows, use the conventional location as it is going to
+help other programs to find Git. RStudio prefers for Git to be installed in 
+"C:/Program Files". Also, when asked about “Adjusting your PATH environment”, 
+select “Git from the command line and also from 3rd-party software”. Otherwise,
+use the defaults parameters.
+
+
+- **macOS users**
+
+On macOS, Git can be installed directly from the shell using this command : 
+
+```{bash}
+git config
```

Comment:
Is this really how to install git, or configure? Will the OSX terminal somehow recognise that git isn't available and install it?

Reply:
I don’t have a Mac so I can’t check but apparently doing that will elicit an offer to install the XCode developer command line tools, which includes git. I guess the terminal recognises indeed that git isn’t available. ´git --version´ would also work